benchmarks/dynamo/common.py

```diff
@@ -2640,10 +2646,15 @@ def warmup(fn, model, example_inputs, mode, niters=5):
             return latency, peak_mem, dynamo_stats
 
         # Cast the model to float16/float32 as necessary
-        model, example_inputs = self.maybe_cast(model, example_inputs)
+        orig_model, example_inputs = self.maybe_cast(model, example_inputs)
 
         # Use distributed wrapping as necessary
-        model = self.deepcopy_and_maybe_parallelize(model)
+        model = self.deepcopy_and_maybe_parallelize(orig_model)
+        if experiment.func is speedup_experiment:
+            # If DDP + compiler is enabled, we need to use a different
+            compiled_model = self.deepcopy_and_maybe_parallelize(orig_model)
```

Comment:
In the future for large models, I suspect we might need to `del orig_model` after deepcopy to get more memory and avoid OOM